src/backend/gporca/libgpopt/include/gpopt/xforms/CXformUtils.h

```diff
@@ -749,12 +749,18 @@ CXformUtils::ImplementHashJoin(CXformContext *pxfctxt, CXformResult *pxfres,
 	CExpressionArray *pdrgpexpr =
 		CCastUtils::PdrgpexprCastEquality(mp, pexprScalar);
 	ULONG ulPreds = pdrgpexpr->Size();
+	BOOL is_hash_join_null_aware = false;
 	for (ULONG ul = 0; ul < ulPreds; ul++)
 	{
 		CExpression *pexprPred = (*pdrgpexpr)[ul];
 		if (CPhysicalJoin::FHashJoinCompatible(pexprPred, pexprOuter,
 											   pexprInner))
 		{
+			if (!is_hash_join_null_aware)
+			{
+				is_hash_join_null_aware |= CPredicateUtils::FINDF(pexprPred);
```

Comment:
Feels like something is redundant; either you don't need `if` statement or you don't need `|`.

```suggestion
			if (!is_hash_join_null_aware)
			{
				is_hash_join_null_aware = CPredicateUtils::FINDF(pexprPred);
```

Or alternatively
```suggestion
			is_hash_join_null_aware |= CPredicateUtils::FINDF(pexprPred);
```

Reply:
Good point. Went with the first.